state/protocol/badger/state.go

```diff
@@ -429,17 +429,18 @@ func bootstrapSealingSegment(
 				return fmt.Errorf("could not index block seal: %w", err)
 			}
 
-			// For all but the first block in the segment, index the parent->child relationship:
-			if i > 0 {
-				err = operation.UpsertBlockChildren(lctx, w, proposal.Block.ParentID, []flow.Identifier{blockID})
```

Comment:
Previously we skip adding index for this block itself that the new block has no child, it works because it will be updated by the next block indexing itself as child of its parent. However, this update is incomplete.

The new approach will always update twice, once for its parent's children, once for its own children (itself has empty child). This takes more updates, but we are only doing it during bootstrapping, which is ok.